src/repository.rs

```diff
@@ -774,6 +1086,131 @@ mod test {
     use futures_executor::block_on;
     use tempfile;
 
+    // Old behavior of the `HttpRepository::get` extension
+    // functionality
+    fn http_repository_extend_using_url(
+        base_url: Url,
+        prefix: &Option<Vec<String>>,
+        components: &[String],
+    ) -> url::Url {
+        let mut url = base_url.clone();
+        {
+            let mut segments = url.path_segments_mut().unwrap();
+            if let Some(ref prefix) = prefix {
+                segments.extend(prefix);
+            }
+            segments.extend(components);
+        }
+        return url;
+    }
+
+    #[test]
+    fn http_repository2_uri_construction() {
+        let base_uri = "http://example.com/one";
+
+        let prefix = Some(vec![String::from("prefix")]);
+        let components = [
+            String::from("componenents_one"),
+            String::from("components_two"),
+        ];
+
+        let uri = base_uri.parse::<Uri>().unwrap();
+        let extended_uri = extend_uri(uri, &prefix, &components).unwrap();
+
+        let url =
+            http_repository_extend_using_url(Url::parse(base_uri).unwrap(), &prefix, &components);
+
+        assert_eq!(url.to_string(), extended_uri.to_string());
+    }
+
+    #[test]
+    fn http_repository2_uri_construction_encoded() {
+        let base_uri = "http://example.com/one";
+
+        let prefix = Some(vec![String::from("prefix")]);
+        let components = [String::from("chars_to_encode<>!")];
+        let uri = base_uri.parse::<Uri>().unwrap();
+        let extended_uri = extend_uri(uri, &prefix, &components)
+            .expect("correctly generated a URI with a zone id");
+
+        let url =
+            http_repository_extend_using_url(Url::parse(base_uri).unwrap(), &prefix, &components);
+
+        assert_eq!(url.to_string(), extended_uri.to_string());
```

Comment:
Can you also assert that these match a fixed string? It makes sense to check that url agrees with you, but it'd also be nice to see what it is that you're expecting while reading the test.

Reply:
Done.